integration-tests/contracts/ethereum_keeper_contracts.go

```diff
@@ -1662,6 +1718,42 @@ func (v *EthereumKeeperRegistrar) EncodeRegisterRequest(
 			amount,
 			common.HexToAddress(senderAddr),
 		)
+
+		if err != nil {
+			return nil, err
+		}
+		return req, nil
+	} else if v.registrar21 != nil {
+		registrarABI, err := abi.JSON(strings.NewReader(registrar21.AutomationRegistrarMetaData.ABI))
+		if err != nil {
+			return nil, err
+		}
+
+		// set default TriggerType to 0(conditional), AutoApproveConfigType to 2(auto approve enabled), AutoApproveMaxAllowed to 1000
+		triggerConfigStruct := registrar21.AutomationRegistrar21InitialTriggerConfig{
+			TriggerType: 0, AutoApproveType: 2, AutoApproveMaxAllowed: 1000,
+		}
+
+		// TODO is this right? Do we need to strip the first 4 bytes?
+		triggerConfigBytes, err := registrarABI.Pack("setTriggerConfig", &triggerConfigStruct)
+
+		// TODO do we need to set offchainConfig in order to get Perform count
+		req, err := registrarABI.Pack(
+			"register",
+			name,
+			email,
+			common.HexToAddress(upkeepAddr),
+			gasLimit,
+			common.HexToAddress(adminAddr),
+			uint8(0), // trigger type
+			checkData,
+			triggerConfigBytes,
+			//[]byte{}, // triggerConfig
+			[]byte{}, // offchainConfig
```

Comment:
upkeeps are not performed. My hunch is this triggerConfig and offchainConfig are not set properly.

Reply:
v21 conditional upkeep e2e is not ready.